var/spack/repos/builtin/packages/zfp/package.py

```diff
@@ -56,15 +59,25 @@ def edit(self, spec, prefix):
             spec.variants['bsws'].value)
 
     def build(self, spec, prefix):
-        make("shared")
+        with working_dir('src'):
+            if '~shared' in spec:
+                make('static')
+            else:
+                make('shared')
 
     def install(self, spec, prefix):
         incdir = 'include' if spec.satisfies('@0.5.1:') else 'inc'
 
+        if self.run_tests:
+            make('test')
+
         # No install provided
         mkdirp(prefix.lib)
         mkdirp(prefix.include)
-        install('lib/libzfp.so', prefix.lib)
+        if '~shared' in spec:
+            install('lib/libzfp.a', prefix.lib)
+        else:
+            install('lib/libzfp.so', prefix.lib)
```

Comment:
While here, can u please use 'dso_suffix' instead?

Reply:
Well, the package builds a `.so` file on osx. I don't plan on changing the package. Many libs I have used wind up building `.so` files on osx and not `.dylib` files. If the package did build a `.dylib`, then I would use dso_suffix. But, it doesn't.

Reply:
if the package uses `.so` I'd stick with that -- but maybe add a comment so that future maintainers know it's ZFP's fault.

Reply:
> but maybe add a comment so that future maintainers know it's ZFP's fault

exactly my point in above https://github.com/LLNL/spack/pull/5212#discussion_r135189929